crates/iceberg/src/arrow/schema.rs

```diff
@@ -385,25 +389,236 @@ impl ArrowSchemaVisitor for ArrowSchemaConverter {
     }
 }
 
+struct ToArrowSchemaConverter;
+
+enum ArrowSchemaOrFieldOrType {
+    Schema(ArrowSchema),
+    Field(Field),
+    Type(DataType),
+}
+
+impl SchemaVisitor for ToArrowSchemaConverter {
+    type T = ArrowSchemaOrFieldOrType;
+
+    fn schema(
+        &mut self,
+        _schema: &crate::spec::Schema,
+        value: ArrowSchemaOrFieldOrType,
+    ) -> crate::Result<ArrowSchemaOrFieldOrType> {
+        let struct_type = match value {
+            ArrowSchemaOrFieldOrType::Type(DataType::Struct(fields)) => fields,
+            _ => unreachable!(),
+        };
+        Ok(ArrowSchemaOrFieldOrType::Schema(ArrowSchema::new(
+            struct_type,
+        )))
+    }
+
+    fn field(
+        &mut self,
+        field: &crate::spec::NestedFieldRef,
+        value: ArrowSchemaOrFieldOrType,
+    ) -> crate::Result<ArrowSchemaOrFieldOrType> {
+        let ty = match value {
+            ArrowSchemaOrFieldOrType::Type(ty) => ty,
+            _ => unreachable!(),
+        };
+        let mut metadata = HashMap::new();
+        metadata.insert(PARQUET_FIELD_ID_META_KEY.to_string(), field.id.to_string());
```

Comment:
Out of curiosity, I was checking how big the hashmap is when initializing, and it turns out it is [empty](https://doc.rust-lang.org/std/collections/struct.HashMap.html#method.new). So maybe good to already set the Field-ID since that one is required?

```suggestion
        let mut metadata = HashMap::HashMap::from([
            (PARQUET_FIELD_ID_META_KEY.to_string(), field.id.to_string()
        ]);
```